components/ILIAS/EventHandling/classes/Definition.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\EventHandling;
+
+class Definition
+{
+    public function __construct(
+        protected string $component,
+        protected string $type,
```

Comment:
Could we make this an `enum` =)?

Reply:
sure :)

Reply:
The type? I guess so, but I'd rather not blow up this PR further right now.